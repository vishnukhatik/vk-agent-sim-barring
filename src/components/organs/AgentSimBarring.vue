<template>
  <div-tag>
    <div class="sim-barring">
                <div class="header">
                  <span class="text-black modal__title">SIM barring</span
                  ><span class="text-black p-x-15px">MSISDN: 491795970587</span>
                </div>
                <div-tag class="p-y-12px" v-if="gpsMsData && isPayment">
                  <payment-method-list
                    :vData="gpsMsData.paymentMethodDetails"
                    :showPaymentMethods="['Direct Debit']"
                    @cardDetails="onCardDetailClick"
                    @passValue="onCheckedButton"
                    :addPaymentMethodUrl="getAemContext.sepaMandateUrl"
                    :disableRadioButton="disableRadioButton"
                  />
                </div-tag>
                <div-tag v-else>
                  <sim-barring />
                </div-tag>
                <div class="u--text-right">
                  <button
                    class="button button-border u--mr-micro-xs u--ml-micro-xs"
                  >
                    Cancel
                  </button>
                  <button
                  v-if="isPayment"
                  @click="isPayment = false"
                    class="button button-border u--mr-micro-xs u--ml-micro-xs"
                  >
                    <!-- {{isPayment ? 'Back' : 'Reset' }} -->
                    Reset
                  </button>
                  <button
                  @click="isPayment = true"
                    class="button button-default u--mr-micro-xs u--ml-micro-xs"
                  >
                    Proceed
                  </button>
                </div>
      <!-- <authenticate @authError="authError" @togglePopup="togglePopup" /> -->
    </div>
  </div-tag>
</template>

<script>
import { DivTag } from "../../../external/ui-kit.esm";
import { SimBarring } from "../molecules";
import { createNamespacedHelpers, mapActions, mapState } from "vuex"; 
const { mapGetters } = createNamespacedHelpers('tefAgentSimBarring')

import { PaymentMethodList } from "../molecules";

export default {
  components: {
    DivTag,
    SimBarring,
    PaymentMethodList
  },
  data() {
    return {
     showModal: false,
     aemStaticData: {},
     disableRadioButton: false,
     isPayment: false
    };
  },
  props: {},
  computed: {
    ...mapState(["AemStaticData","gpsData","customerID","events",]),
    ...mapGetters([
      // 'getPaymentMethods',
      'gpsMsData',
      'getAEMData',
      'getAemContext'
    ]),
  },
  mounted() {},
  created() {},
  methods: {
    ...mapActions('tefAgentSimBarring', [
      'setCurrentStepData',
      'onCardDetailClick',
    ]),

    async submitOrder() {

      //DRC Call
      const systemDate = new Date()

      const requestParams = {
        url: this.getAemContext.drcMsUrl,
        payload: {
          msisdn: this.getPhoneNumber, // this.msisdnValues[0].value,
          paymentMethodReference: this.getPaymentMethod.id,
          amount: this.getSelectedReason.rate,
          currency: 'EUR',
          paymentDate: systemDate,
          paymentUseCase: 'ON_DEMAND',
        },
      }
      
      await this.getDrcMsData(requestParams)
       if (this.drcData.data.status === 400 || this.drcData.data.errorMessage) {
          //this.errorMessages = this.drcData.data.errorMessage
           this.$store.commit("SET_API_ERROR_MSG", 'Unfortunately, your SIM swap request was unsuccessful, we are working on it. We will keep you updated.');
          //  this.$store.dispatch("setCurrentStepData", {
          //   currentStep: "apiFail"
          // });
          this.dispatchEvent(this.events.SimSwapOrderFailed, {});


          
      } else if (this.drcData.data) { 
       
          if(await this.$store.dispatch("submitOrder", this.getAemContext) ){
              this.showModal = true;
              setTimeout(() => {
                  this.showModal = false;
                  this.dispatchEvent(this.events.SimSwapOrderSuccess, {});
              }, 2000);
          }
          else{
             this.$store.commit("SET_API_ERROR_MSG", ' Unfortunately, your SIM swap request was unsuccessful, there is issue in order submission, we are working on it. We will keep you updated.');
            //  this.$store.dispatch("setCurrentStepData", {
            //   currentStep: "apiFail"
            // });
            this.dispatchEvent(this.events.SimSwapOrderFailed, {});
          } 
      }
    },


    onCardDetailClick(payload) {
      this.$emit('onCardDetails', payload);
    },
    onCheckedButton(e) {
      this.$emit('checkInputValue', e);
    }

  },
};
</script>

<style scoped lang="scss"></style>
