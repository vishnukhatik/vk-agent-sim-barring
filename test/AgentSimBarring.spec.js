import { shallowMount, createLocalVue } from "@vue/test-utils";
import AgentSimBarring from "../src/components/organs/AgentSimBarring.vue";
import Vuex from "vuex";

const localVue = createLocalVue();

localVue.use(Vuex);

let wrapper;
let store;
let actions;
let mutations;
let getters;
let state;

beforeEach(() => {
  jest.clearAllMocks()
  actions = {};
  mutations = {};
  state = {
    key: {},
  };
  getters = {
    gpsMsData: jest.fn(),
  };
  store = new Vuex.Store({
    strict: true,
    modules: {
      tefAgentSimBarring: {
        namespaced: true,
        state,
        getters,
        mutations,
        actions,
      },
    },
  });

  wrapper = shallowMount(AgentSimBarring, {
    propsData: {},
    store,
    localVue,
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe("AgentSimBarring.vue", () => {
  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("is a instance of SimBarring", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  test("is a Vue instance", () => {
    expect(wrapper.isVueInstance).toBeTruthy();
  });
});
