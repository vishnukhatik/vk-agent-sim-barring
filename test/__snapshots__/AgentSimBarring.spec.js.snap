// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentSimBarring.vue matches snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
